src/Expectations/OppositeExpectation.php

```diff
@@ -29,15 +30,19 @@ public function __construct(private Expectation $original)
     /**
      * Asserts that the value array not has the provided $keys.
      *
-     * @param array<int, int|string> $keys
+     * @param array<int, int|string|array<int[]|string[]>> $keys
```

Comment:
For a recursive array, this could be more readable by phpstan:


```suggestion
     * @param array<int, int|string|array<int-string, mixed>> $keys
```

Otherwise you could have a phpstan error on this:

`expect($foo)->toHaveKeys(['bar' => ['baz' => ['quuz', 'quz']]])`

As it will complain that the third array level should be made of strings or ints

Reply:
OK, I corrected the both files.